lumigator/python/mzai/backend/backend/services/experiments.py

```diff
@@ -0,0 +1,48 @@
+import json
+import time
+
+import loguru
+import requests
+from fastapi import BackgroundTasks
```

Comment:
IMHO we should get used to handling async and the event loop directly. For example, it seems the source for BackgroundTasks handles a threadpool for sync funs and asyncio for async funs. In our case, we may find clogging a threadpool just by running sleeps, which are other supported in a single thread when using asyncs. Inside the backend, we're not yet running cpu intensive tasks that require using separate threads.

Reply:
Thank you! I think my answer below addresses this comment too re: both BackgroundTasks vs manual implementation and threads vs async, but please lmk if there's still something not clear!

I appreciate your comment and will make sure this is addressed in the inference job PR where ray client is already used. 🙏